File: Code/SimpleBudget.API/Models/ProductPrices/ProductPriceMultiModel.cs
using System.ComponentModel.DataAnnotations;

namespace SimpleBudget.API.Models
{
    public class ProductPriceMultiModel
    {
        public class PriceModel
        {
            [Required]
            public string ProductName { get; set; } = default!;

            [Required]
            public int Quantity { get; set; }

            [Required]
            public decimal Price { get; set; }

            [Required]
            public bool IsDiscount { get; set; }

            public string? Description { get; set; }
        }

        public string? CompanyName { get; set; }
        public string? CategoryName { get; set; }

        [Required]
        public string PriceDate { get; set; } = default!;

        [Required]
        public PriceModel[] Prices { get; set; } = default!;
    }
}

File: Code/client/src/api/models/ProductPriceMultiModel.ts
interface PriceModel
{
    productName: string;
    quantity: number;
    price: number;
    isDiscount: boolean;
    description: string | undefined | null;
}

export interface ProductPriceMultiModel {
    companyName: string | undefined | null;
    categoryName: string | undefined | null;
    priceDate: string;
    prices: PriceModel[];
}
